Mathlib/Data/Fin/Tuple/NatAntidiagonal.lean

```diff
@@ -155,14 +155,10 @@ theorem antidiagonalTuple_pairwise_pi_lex :
       rw [antidiagonal_zero]
       exact List.pairwise_singleton _ _
     | succ n n_ih =>
-      rw [antidiagonal_succ, List.pairwise_cons, List.pairwise_map]
-      refine ⟨fun p hp x hx y hy => ?_, ?_⟩
-      · rw [List.mem_map, Prod.exists] at hp
-        obtain ⟨a, b, _, rfl : (Nat.succ a, b) = p⟩ := hp
-        exact Or.inl (Nat.zero_lt_succ _)
-      dsimp
-      simp_rw [Nat.succ_inj, Nat.succ_lt_succ_iff]
-      exact n_ih
+      simp only [antidiagonal_succ, pairwise_cons, mem_map, mem_antidiagonal, Prod.exists,
+        Prod.map_apply, Nat.succ_eq_add_one, id_eq, forall_exists_index, and_imp, Prod.forall,
+        Prod.mk.injEq]
```

Comment:
This is just the result of a bare `simp?` call, right? If so, because it precedes `grind`, I think it's okay to leave it as `simp`. The flexible tactic linter shouldn't complain because `grind` is also flexible.

If the list of lemmas were smaller, I might not recommend it, but large `simp only` calls can cause additional maintainence burden (e.g., when lemmas are renamed).